doc/tutorials/snippet-sharing-rest-service.md

```diff
@@ -0,0 +1,196 @@
+
+- Add APT repository: `add-apt-repository ppa:ben-collins/libjwt`
+- Install via apt-get: `apt-get update && apt-get install libjwt`
+
+#### Frontend ####
+
+The frontend does only require the package manager [npm](https://www.npmjs.com/).
+It is preferred to install it along with [nodeJS](https://nodejs.org/).
+
+To install npm via APT, use: `apt-get install npm`
+
+### Build the Applications ###
+
+After installing the dependencies, we are ready to build the applications.
+To do so, we can follow the steps explained in the [build guide](/doc/COMPILE.md).
+Make sure to include the two tools `rest-backend` and `rest-frontend`, e.g. by
+using the arguments `-DTOOLS=ALL` or `-DTOOLS=rest-backend;rest-frontend`.
+
+## Configuration ##
+
+The applications configure themselves as much as possible during build,
+but some settings have to be set manually afterwards.
+
+### Backend ###
+
+To make sure that the configuration specification was set properly during installation,
+use the command `kdb mount`. You should see a list of mountpoints, of which one should
+be something like `/sw/elektra/restbackend/#0`. If you do not see this mountpoint,
+use `kdb mount rest-backend-spec.ini /sw/elektra/restbackend/#0 ni` to mount it manually.
+
+After that you need to set three additional configuration parameters that have no defaults.
+It is recommended to set them for the system namespace if you will use a tool like
+`systemctl` to manage the services. (For the api_specification key, see below!)
+```
+kdb set system/sw/elektra/restbackend/#0/current/backend/jwt/encryption_key "use_a_secret_key_here!"
+kdb set system/sw/elektra/restbackend/#0/current/backend/api_specification/html "http://link.to/the/html/version/of/api/description"
+kdb set system/sw/elektra/restbackend/#0/current/backend/api_specification/raw "http://link.to/the/blueprint/of/api/description"
+```
+
+In case you don't want to publish the API description, you can also set the keys
+to other links, e.g. your main page. It is good practice to have the description
+published though.
+
+To change the port of the server to 12345, you can use:
+`kdb set system/sw/elektra/restbackend/#0/current/cppcms/service/port 12345`
+
+### Frontend ###
+
+The frontend does only require small mandatory changes, which have to be made in the
+`application-config.json` in the `/usr/local/share/elektra/tool_data/rest-frontend`
+directory:
+
+- Change `backend.root` to the URL where the backend will be reachable,
+  e.g. `http://restapi.libelektra.org/` (with trailing slash!)
+- Change `website.url` to the URL where the frontend will be reachable,
+  e.g. `http://libelektra.org/` (with trailing slash!)
+
+## Running the Applications ##
+
+As last step we need to run the applications:
+
+- First we start the backend server with `kdb run-rest-backend`.
+  To ensure the backend is accessible, you can use
+  `curl http://localhost:8080/version` (change port to your setting),
```

Comment:
We just set the port 10 lines above - I hope no one forgets it that fast. :smiley: